tests/spectest/spec_test.go

```diff
@@ -196,79 +197,80 @@ func (c command) expectedError() (err error) {
 }
 
 func addSpectestModule(t *testing.T, store *wasm.Store) {
-	memoryLimitMax := uint32(2)
+	// See https://github.com/WebAssembly/spec/blob/wg-1.0/test/core/imports.wast
+	mod, err := text.DecodeModule([]byte(`(module $spectest
+(; TODO
+  (global (export "global_i32") i32)
+  (global (export "global_i64") i64)
+  (global (export "global_f32") f32)
+  (global (export "global_f64") f64)
+
+  (table (export "table") 10 20 funcref)
+;)
+
+;; TODO: revisit inlining after #215
+
+  (memory 1 2)
+    (export "memory" (memory 0))
+
+;; Note: the following aren't host functions that print to console as it would clutter it. These only drop the inputs.
+  (func)
+     (export "print" (func 0))
+
+  (func (param i32) local.get 0 drop)
+     (export "print_i32" (func 1))
+
+  (func (param i64) local.get 0 drop)
+     (export "print_i64" (func 2))
+
+  (func (param f32) local.get 0 drop)
+     (export "print_f32" (func 3))
+
+  (func (param f64) local.get 0 drop)
+     (export "print_f64" (func 4))
+
+  (func (param i32 f32) local.get 0 drop local.get 1 drop)
+     (export "print_i32_f32" (func 5))
+
+  (func (param f64 f64) local.get 0 drop local.get 1 drop)
+     (export "print_f64_f64" (func 6))
+)`), wasm.Features20191205, wasm.MemoryMaxPages)
+	require.NoError(t, err)
+
+	// (global (export "global_i32") i32)
+	mod.GlobalSection = append(mod.GlobalSection, &wasm.Global{
+		Type: &wasm.GlobalType{ValType: wasm.ValueTypeI32},
+		Init: &wasm.ConstantExpression{Opcode: wasm.OpcodeI32Const, Data: []byte{0x9a, 0x5} /* = 666 */},
+	})
+	mod.ExportSection["global_i32"] = &wasm.Export{Name: "global_i32", Index: 0, Type: wasm.ExternTypeGlobal}
+
+	// (global (export "global_i64") i64)
+	mod.GlobalSection = append(mod.GlobalSection, &wasm.Global{
+		Type: &wasm.GlobalType{ValType: wasm.ValueTypeI64},
+		Init: &wasm.ConstantExpression{Opcode: wasm.OpcodeI64Const, Data: []byte{0x9a, 0x5} /* = 666 */},
+	})
+	mod.ExportSection["global_i64"] = &wasm.Export{Name: "global_i64", Index: 1, Type: wasm.ExternTypeGlobal}
+
+	// (global (export "global_f32") f32)
+	mod.GlobalSection = append(mod.GlobalSection, &wasm.Global{
+		Type: &wasm.GlobalType{ValType: wasm.ValueTypeF32},
+		Init: &wasm.ConstantExpression{Opcode: wasm.OpcodeF32Const, Data: []byte{0x44, 0x26, 0x80, 0x00}},
```

Comment:
ps I'm not sure in the docs where the magic constants are defined.. if you know, please let me and I'll comment them.

Reply:
or if we define them, I can just comment "arbitrary"

Reply:
I thought the const is here https://github.com/WebAssembly/spec/blob/ffb5e3b40db7a45237d4e2fc972c1552ac4abef5/interpreter/README.md#spectest-host-module but I couldn't find :D but they need to be these values otherwise fail so not arbitrary :D

Reply:
oh found it https://github.com/WebAssembly/spec/blob/f5a260a2025ba4d7d398654581c7c532e3a2c319/interpreter/script/js.ml#L44-L49

Reply:
I found them https://github.com/WebAssembly/spec/blob/wg-1.0/test/harness/sync_index.js#L79-L91

always 🕵️